etl/steps/__init__.py

```diff
@@ -499,6 +499,13 @@ def checksum_input(self) -> str:
         for f in self._step_files():
             checksums[f] = files.checksum_file(f)
 
+            # if using SUBSET to process just a subset of the data, then add it to its checksum if
+            # the file contains it
+            if config.SUBSET:
+                with open(f) as istream:
+                    if "SUBSET" in istream.read():
+                        checksums[f] += config.SUBSET
+
```

Comment:
This is nice, so that a different subset choice (or the lack of a subset) triggers a rebuild 👍